### PR TITLE
III-3473 Production id filter

### PR DESCRIPTION
**Pull request by bertramakers**
### Added
 
- Added production id filter
 
---

Ticket: https://jira.uitdatabank.be/browse/III-3473

---

Todo:

- [X] Add field mapping for `production.id`
- [X] Increase `UDB3_CORE` version number
- [X] Add logic to copy production id from incoming json-ld to elasticsearch json document
- [x] Add `withProductionId` filter to `OfferQueryBuilderInterface`
- [x] Implement `withProductionId` in `ElasticSearchOfferQueryBuilder`
- [x] Add request parser for `productionId` URL parameter
- [x] Inject new request parser in `OfferSearchController`

### CODE REVIEW

**Comment by bertramakers** [comment]
This is ready for review but will need the fixes in #68 to make the migration work correctly.

File: src/ElasticSearch/Event/CopyJsonEvent.php [hunk 1]
```diff
@@ -63,6 +71,10 @@ public function copy(\stdClass $from, \stdClass $to)
 
         $this->copyJsonRelatedLocation->copy($from, $to);
 
+        $this->copyRelatedProduction->copy($from, $to);
+
+        // Needs to go last for a fallback to calculate the completedLanguages based on the properties on the new
+        // document if it's missing in the original.
```
Comment (jdrieghe) [inline src/ElasticSearch/Event/CopyJsonEvent.php:77]: Thanks for the comment 👏

File: src/Http/OfferSearchController.php [hunk 2]
```diff
@@ -126,7 +126,7 @@ public function __construct(
         $this->regionDocumentType = $regionDocumentType;
         $this->queryStringFactory = $queryStringFactory;
         $this->facetTreeNormalizer = $facetTreeNormalizer;
-        $this->offerParameterWhiteList = new OfferParameterWhiteList();
+        $this->offerParameterWhiteList = new OfferSupportedParameters();
```
Comment (jdrieghe) [inline src/Http/OfferSearchController.php:129]: 🎖️

**Review by jdrieghe — APPROVED** [review]

File: src/Http/OfferSearchController.php (hunk 2 above)
Comment (bertramakers) [reply src/Http/OfferSearchController.php:129]: It's actually also clearer imo 😄

**Pull request merged**